### PR TITLE
Add metacognitive training, Socratic refinement, and memory-safety logging

### PR DESCRIPTION
**Pull request by InfiniteMalice**
## Problem statement

This PR adds opt-in, modular scaffolding for memory-mediated semantic laundering protection, CPT-inspired metacognitive pairwise training, SSR-inspired step-level refinement, and a structured logging contract for longitudinal reflective-stability analysis.

## Existing functionality preserved

- Existing semantic-intent analysis and `run_conversation(...)` remain separate from memory safety.
- Reward decomposition is preserved, including positive-only thought-alignment behavior.
- New runtime behavior is disabled by default through feature flags.
- Legacy trace rows continue loading in the viewer.

## Memory-mediated laundering safeguards

- Adds `modules/semantic_intent_robustness/memory_safety.py` with typed source/trust/durability/write/retrieval decisions.
- Rejects protected-policy/constitutional/system-state override attempts.
- Quarantines untrusted durable instruction promotion, missing provenance, identity/authority/goal changes, delayed activation, unsafe tool-selection influence, and high-risk durable writes.
- Adds `SemanticIntentPipeline.run_memory_boundary(...)` as a separate boundary method.

## CPT-inspired mid-training implementation

- Adds `modules/cognitive_pairwise_training/` with typed reasoning-trace candidates, pairwise examples, labels, pair types, deterministic pair construction, JSONL export, and trainer-facing diagnostics.
- Supports intra-model, inter-model, smaller-correct/larger-incorrect, answer-process mismatch, same-answer/different-quality, abstention-vs-guess, semantic-laundering, and principle-pressure stress pairs.
- This is labeled as lightweight scaffolding, not full production fine-tuning.

## SSR-inspired refinement implementation

- Adds `modules/socratic_self_refine/` with reasoning units, controlled resolve attempts, assessments, repair events, bounded run reports, and evaluation metrics.
- Preserves original traces, bounds iterations, supports policy checks before/after repair, and escalates review when policy-relevant meaning changes.
- Disabled by default.

## Logging-contract changes

- Adds `src/mindful_trace_gepa/logging_schema.py` with `EventEnvelope`, event types, normalization helpers, and trainer metric optional-field support.
- Extends trainer metrics with optional confidence, abstention, trace summary, reward components, metadata, and external references.
- Token telemetry now labels synthetic lightweight telemetry explicitly.
- Circuit telemetry now uses `null` plus status metadata when unavailable, rather than measured zero.

## Viewer changes

- Viewer builder normalizes event-envelope rows while preserving legacy rows.
- Viewer displays event types, event IDs, payloads, and telemetry availability labels.
- Adds event-type filtering.

## Backward-compatibility strategy

- Feature flags default new behavior off.
- Existing trace rows still load.
- Existing 17-case/schema files were not modified.
- YAML imports are optional at import time, with a fallback for empty/simple config files in lightweight environments.

## Feature flags and defaults

Added defaults for `memory_safety`, `cognitive_pairwise_training`, `socratic_self_refine`, and `structured_logging` in config examples and training config dataclasses. CPT/SSR/memory safety are disabled by default.

## Tests added

- `tests/test_memory_mediated_laundering.py`
- `tests/test_cognitive_pairwise_training.py`
- `tests/test_socratic_self_refine.py`
- `tests/test_structured_logging_contract.py`

## Exact validation commands and results

Passed:

```bash
C:\Users\evanh\.cache\codex-runtimes\codex-primary-runtime\dependencies\python\python.exe -m pytest tests/test_memory_mediated_laundering.py tests/test_cognitive_pairwise_training.py tests/test_socratic_self_refine.py tests/test_structured_logging_contract.py tests/test_viewer_build.py tests/test_training_configs.py tests/test_training_cli_logging.py -q
# 37 passed in 3.12s

C:\Users\evanh\.cache\codex-runtimes\codex-primary-runtime\dependencies\python\python.exe -m pytest modules/semantic_intent_robustness/tests -q
# 47 passed in 0.39s

C:\Users\evanh\.cache\codex-runtimes\codex-primary-runtime\dependencies\python\python.exe -m compileall modules\semantic_intent_robustness\memory_safety.py modules\cognitive_pairwise_training modules\socratic_self_refine src\mindful_trace_gepa\logging_schema.py src\mindful_trace_gepa\cli.py src\mindful_trace_gepa\tokens.py src\mindful_trace_gepa\viewer\builder.py gepa_mindfulness\training\base_trainer.py gepa_mindfulness\training\config.py gepa_mindfulness\training\configs.py gepa_mindfulness\core\tracing.py tests\test_memory_mediated_laundering.py tests\test_cognitive_pairwise_training.py tests\test_socratic_self_refine.py tests\test_structured_logging_contract.py
# passed
```

Attempted but blocked by missing tools/dependencies in the bundled runtime:

```bash
pytest -q
# collection blocked: missing yaml and torch in runtime

python -m ruff check .
# No module named ruff

python -m mypy .
# No module named mypy
```

## Known limitations

- CPT is a deterministic data/diagnostic scaffold, not an integrated model fine-tuning loop.
- SSR controlled resolving is an inspectable local scaffold; production inference integrations should provide real verifier/model callbacks.
- Viewer support is additive and compact; it does not redesign the offline UI.
- Full test suite could not run in this environment because required optional dependencies were unavailable.

## Follow-up work

- Attach CPT batches to the live pre-RL training stage when a model training backend is available.
- Add production verifier/model callbacks for SSR controlled re-solving.
- Expand structured event emission at more rollout/trainer integration points.
- Run full CI with repository dependencies installed.

## 17-case confirmation

The 17-case schema was not expanded, renumbered, or semantically redefined. The relevant 17-case/schema files have no diff in this PR.

## Summary by CodeRabbit

* **New Features**
  * Cognitive Pairwise Training (CPT) and Socratic Self-Refine (SSR) pipelines and CLI commands (cpt-build, ssr-run)
  * Memory-safety boundary checks and reporting
  * Structured event logging with envelopes, schema/versioning, and telemetry fields
  * Viewer: timeline event-type filtering and richer telemetry display

* **Improvements**
  * Config toggles for robustness/safety/observability; YAML optional with JSON fallback
  * Enhanced training metric records and token/circuit telemetry labeling

* **Tests & Docs**
  * New tests and documentation for CPT, SSR, memory safety, and structured logging

### CODE REVIEW

**Comment by InfiniteMalice** [comment]
Follow-up pushed in `927dc9e`.

Addressed review feedback:

- `compute_pairwise_label_loss(...)` now requires `predicted_probabilities` length to match `examples` when predictions are supplied, preventing silent truncation/extra-prediction masking from `zip(...)`.
- Viewer event filtering now selects details from the filtered visible event list instead of calling `selectEvent(0)` on unfiltered `pageEvents`. Applied to both `viewer.js` and the `.new` asset that the builder prefers.
- CPT/SSR scaffolds are now grounded in the existing cooperation-under-uncertainty synthetic generator via `generate_cooperation_cpt_candidates()` and `generate_cooperation_ssr_units()`, with tests covering both.
- Added `dev` optional dependencies for `black`, `ruff`, `mypy`, `pytest`, and `pyyaml`; installed missing tools in the local bundled runtime for validation.

Validation after follow-up:

```bash
python -m black --check .
# 387 files would be left unchanged

python -m ruff check .
# All checks passed

python -m mypy modules/cognitive_pairwise_training synthetic_data/generators/cooperation_under_uncertainty_generator.py modules/socratic_self_refine
# Success: no issues found in 7 source files

python -m pytest tests/test_memory_mediated_laundering.py tests/test_cognitive_pairwise_training.py tests/test_socratic_self_refine.py tests/test_structured_logging_contract.py tests/test_reality_contact_scaffolding.py tests/test_viewer_build.py tests/test_training_configs.py tests/test_training_cli_logging.py -q
# 46 passed in 2.86s

python -m pytest modules/semantic_intent_robustness/tests -q
# 47 passed in 0.25s
```

Full `pytest -q` still cannot collect in this local runtime because `torch` is not installed (`tests/test_dapo_hybrid_trainer.py`).
